lib/goto-definition.coffee

```diff
@@ -66,7 +66,17 @@ module.exports =
     scan_regex = scan_regex.filter (e, i, arr) -> arr.lastIndexOf(e) is i
     scan_paths = scan_paths.filter (e, i, arr) -> arr.lastIndexOf(e) is i
 
-    regex = scan_regex.join('|').replace(/{word}/g, word)
+    # Word recognition doesn't always work perfectly (e.g. clojure deref with
```

Comment:
maybe we could make some different `getSelectedWord` to the language. in clojure, we select keyword with `/[$0-9\w-]+/`, and in perl6 we use `/[$0-9\w-\$\@\\\%]+/`, the regex is general. is this okay ?